### PR TITLE
[FLINK-10010][DataStream API] Deprecate unused BaseAlignedWindowAssigner related components

### PR DESCRIPTION
**Pull request by walterddr**
## What is the purpose of the change

Deprecate unused AlignedWindow related components. the links in comments are all dead and WindowOperator already decline assigners of such type.

## Brief change log

  - Mark Deprecated BaseAlignedWindowAssigner
  - Remove SlidingAlignedProcessingTimeWindows

## Verifying this change

N/A

## Does this pull request potentially affect one of the following parts:

N/A

## Documentation

N/A

### CODE REVIEW

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/windowing/assigners/BaseAlignedWindowAssigner.java [hunk 1]
```diff
@@ -27,15 +27,13 @@
...
- * {@link org.apache.flink.streaming.runtime.operators.windowing.AccumulatingProcessingTimeWindowOperator
- * AccumulatingProcessingTimeWindowOperator} and
- * {@link org.apache.flink.streaming.runtime.operators.windowing.AggregatingProcessingTimeWindowOperator
- * AggregatingProcessingTimeWindowOperator}.
+ * A base {@link WindowAssigner} used to instantiate one of the deprecated operators.
  *
- * <p>For assigner that extend this one, the user can check the
- * {@link TumblingAlignedProcessingTimeWindows} and the {@link SlidingAlignedProcessingTimeWindows}.
+ * @deprecated will be removed in a future version. please use other {@link WindowAssigner}s listed under
+ *  * {@link org.apache.flink.streaming.api.windowing.assigners}.
```
Comment (yanghua) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/windowing/assigners/BaseAlignedWindowAssigner.java]: extra '*'

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/windowing/assigners/BaseAlignedWindowAssigner.java [hunk 2]
```diff
@@ -27,15 +27,13 @@
...
  * */
```
Comment (yanghua) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/windowing/assigners/BaseAlignedWindowAssigner.java]: replace '* */' to '*/'

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/windowing/assigners/BaseAlignedWindowAssigner.java [hunk 3]
```diff
@@ -27,15 +27,13 @@
...
+
```
Comment (yanghua) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/windowing/assigners/BaseAlignedWindowAssigner.java]: please remove this empty line

**Comment by walterddr** [comment]
thx for the review @yanghua . for some reason java checkstyle did not catch these problems. just fixed it.

**Comment by walterddr** [comment]
Hi @fhueske @aljoscha, could you kindly take a look at this PR? We would like to make some changes to the "align" window assigner concept. It would be nice to depreciate current implementation to avoid confusions.

**Comment by aljoscha** [comment]
Thanks @walterddr! I merged this.

**Pull request closed without merging**